llvm/test/CodeGen/AIE/aie2/GlobalISel/prelegalizercombiner-shufflevector-buildvector.mir

```diff
@@ -0,0 +1,322 @@
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(<8 x s32>) = COPY $wl4
+    ; CHECK-NEXT: [[SHUF:%[0-9]+]]:_(<16 x s32>) = G_SHUFFLE_VECTOR [[COPY]](<8 x s32>), [[BUILD_VECTOR]], shufflemask(12, 13, 11, 15, 9, 9, 8, 12, 8, 8, 13, 12, 12, 10, 8, 9)
+    ; CHECK-NEXT: $x0 = COPY [[SHUF]](<16 x s32>)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $x0
+    %0:_(s32) = G_CONSTANT i32 28
+    %1:_(s32) = G_CONSTANT i32 24
+    %2:_(s32) = G_CONSTANT i32 18
+    %3:_(s32) = G_CONSTANT i32 8
+    %4:_(s32) = G_CONSTANT i32 7
+    %5:_(s32) = G_CONSTANT i32 3
+    %6:_(s32) = G_CONSTANT i32 2
+    %7:_(s32) = G_CONSTANT i32 1
+    %8:_(<8 x s32>) = G_BUILD_VECTOR %0(s32), %1(s32), %2(s32), %3(s32), %4(s32), %5(s32), %6(s32), %7(s32)
+    %9:_(<8 x s32>) = COPY $wl4
+    %10:_(<16 x s32>) = G_SHUFFLE_VECTOR %9:_(<8 x s32>), %8:_, shufflemask(12, 13, 11, 15, 9, 9, 8, 12, 8, 8, 13, 12, 12, 10, 8, 9)
+    $x0 = COPY %10:_(<16 x s32>)
+    PseudoRET implicit $lr, implicit $x0
+...
+---
+name:            shufflevector_undef_bv
+legalized:       false
+body:             |
+  bb.1.entry:
+    ; CHECK-LABEL: name: shufflevector_undef_bv
+    ; CHECK: [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 24
+    ; CHECK-NEXT: [[C1:%[0-9]+]]:_(s32) = G_CONSTANT i32 18
+    ; CHECK-NEXT: [[C2:%[0-9]+]]:_(s32) = G_CONSTANT i32 8
+    ; CHECK-NEXT: [[C3:%[0-9]+]]:_(s32) = G_CONSTANT i32 7
+    ; CHECK-NEXT: [[C4:%[0-9]+]]:_(s32) = G_CONSTANT i32 3
+    ; CHECK-NEXT: [[C5:%[0-9]+]]:_(s32) = G_CONSTANT i32 2
+    ; CHECK-NEXT: [[C6:%[0-9]+]]:_(s32) = G_CONSTANT i32 1
+    ; CHECK-NEXT: [[BUILD_VECTOR:%[0-9]+]]:_(<16 x s32>) = G_BUILD_VECTOR [[C3]](s32), [[C4]](s32), [[C2]](s32), [[C6]](s32), [[C]](s32), [[C]](s32), [[C1]](s32), [[C3]](s32), [[C3]](s32), [[C5]](s32), [[C4]](s32), [[C3]](s32), [[C3]](s32), [[C1]](s32), [[C5]](s32), [[C]](s32)
+    ; CHECK-NEXT: $x0 = COPY [[BUILD_VECTOR]](<16 x s32>)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $x0
+    %0:_(s32) = G_CONSTANT i32 28
+    %1:_(s32) = G_CONSTANT i32 24
+    %2:_(s32) = G_CONSTANT i32 18
+    %3:_(s32) = G_CONSTANT i32 8
+    %4:_(s32) = G_CONSTANT i32 7
+    %5:_(s32) = G_CONSTANT i32 3
+    %6:_(s32) = G_CONSTANT i32 2
+    %7:_(s32) = G_CONSTANT i32 1
+    %8:_(<8 x s32>) = G_BUILD_VECTOR %0(s32), %1(s32), %2(s32), %3(s32), %4(s32), %5(s32), %6(s32), %7(s32)
+    %9:_(<8 x s32>) = G_IMPLICIT_DEF
+    %10:_(<16 x s32>) = G_SHUFFLE_VECTOR %9:_(<8 x s32>), %8:_, shufflemask(12, 13, 11, 15, 9, 9, 10, 12, 12, 14, 13, 12, 12, 10, 14, 9)
+    $x0 = COPY %10:_(<16 x s32>)
+    PseudoRET implicit $lr, implicit $x0
+...
+---
+name:            shufflevector_undef_undef
+legalized:       false
+body:             |
+  bb.1.entry:
+    ; CHECK-LABEL: name: shufflevector_undef_undef
+    ; CHECK: [[DEF:%[0-9]+]]:_(<16 x s32>) = G_IMPLICIT_DEF
+    ; CHECK-NEXT: $x0 = COPY [[DEF]](<16 x s32>)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $x0
+    %0:_(s32) = G_CONSTANT i32 28
```

Comment:
Nit: unused G_CONSTANTs here and in the tests below

Reply:
Fair, thanks Konstantin.